pkg/duck/subscriber.go

```diff
@@ -85,40 +82,45 @@ func SubscriberSpec(ctx context.Context, dynamicClient dynamic.Interface, namesp
 		return *s.DeprecatedDNSName, nil
 	}
 
-	obj, err := ObjectReference(ctx, dynamicClient, namespace, s.Ref)
-	if err != nil {
-		logging.FromContext(ctx).Warn("Failed to fetch SubscriberSpec target",
-			zap.Error(err),
-			zap.Any("subscriberSpec.Ref", s.Ref))
-		return "", err
-	}
-
-	if err = track(*s.Ref); err != nil {
+	if err := track(*s.Ref); err != nil {
```

Comment:
we check for nil on L92 (s.Ref != nil), does track and other ListerFor / InformerFor do the right things with a nil or empty?

Reply:
That track is the same as the one that we had before. It receives an objectRef, not a *objectRef, so it cannot be nil.
Followed the same pattern for the ListerFor and InformerFor... i.e., they receive an objectRef.

What do you mean by empty --> objRef.Name == "" or objRef.Namespace == "" ? 
I think the ListerFor and InformerFor shouldn't break, they won't find the lister or informer, and return an error...

We are using 
```	gvk := ref.GroupVersionKind()
	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
```
if empty, it seems that those methods don't break...

Reply:
Synced offline with @vaikas-google. Agreed to add a check to ensureTracking to return an error if the objectRef is empty. Thanks for the comment, made a lot of sense now :)

Reply:
Adding the same check in ListerFor and InformerFor. We avoid grabbing a lock by returning early in such cases.